Lesson1/lesson_1/functions.cpp

```diff
@@ -0,0 +1,76 @@
+#include <vector>
+#include <iostream>
+#include <list>
+#include <forward_list>
+
+//print vector
+void print_vector (const std::vector<int> vector)
+{
+    for(auto &i : vector)
+    {
+        std::cout << i;
+    }
+    std::cout << std::endl;
+}
+
+//print list
+void print_list (const std::list<int> list)
+{
+    for(auto &i : list)
+    {
+        std::cout << i;
+    }
+    std::cout << std::endl;
+}
+
+//print forward_list
+void print_forward_list (const std::forward_list<int> forward_list)
+{
+    for(auto &i : forward_list)
+    {
+        std::cout << i;
+    }
+    std::cout << std::endl;
+}
+
+//task_1
+std::vector<int> concat (const std::vector<int> vector_1, const std::vector<int> vector_2)
+{
+    std::vector<int> tmp_vector {vector_1};
+    tmp_vector.insert(tmp_vector.end(), vector_2.begin(), vector_2.end());
+    return tmp_vector;
+}
+
+//task_2
+std::list<int> Mylist (const int &N, const int &X)
```

Comment:
Usually, basic types are passed by copy to functions, and objects are passed by reference.
So, here, copy is preferable.